vector-im/element-android
# Debug Icon UX Improvements & Addition in New Layout

## Type of change


## Content

- Debug icon now refreshes visibility in onResume so you don't have to restart the app after enabling Developer Mode
- Debug icon added to New App Layout
- New App Layout flag changes take place without having to restart

## Motivation and context

Closes https://github.com/vector-im/element-android/issues/6867

## Screenshots / GIFs

<img width="300" alt="image" src="https://user-images.githubusercontent.com/20701752/185345327-c53ecf80-331d-4a6e-b5b0-2a14bf4dab88.png">

## Tests

- Enable developer mode
- Without restarting, go back to home screen and see debug icon
- Go to debug icon > features > enable new app layout
- Go back without restarting and see new app layout
- See that new app layout also has debug icon in the toolbar
- Disable new app layout and without restarting see that you're back in the old app layout

## Tested devices

- OS version(s): Android 12

## Checklist
